efi/meson.build

```diff
@@ -91,6 +91,11 @@ else
         coff_header_in_crt0 = false
 endif
 
+# For NX compat, we must ensure we have .note.GNU-stack
+if run_command('grep', '-q', '.note.GNU-stack', join_paths(efi_crtdir, arch_crt), check: false).returncode() != 0
+    error('Cannot find NX section in @0@, update to gnu-efi 3.0.15+'.format(join_paths(efi_crtdir, arch_crt)))
+endif
```

Comment:
could this be done more cleanly by the python script we use to set the NX bit perhaps?

Reply:
generate-binary is too late in the build process as otherwise arm will fallback and trigger a false positive at that stage

Reply:
Got it.

Perhaps would we be able to change it so that if command failed to fall back to the files included in tree?  That should help the Fedora CI failure (and probably in the wild on other things that use this too).

Reply:
Still nope as everything has to be built NX for NX to work, the other option is to build a way to disable NX

Reply:
Ah thanks for confirming.  So gnu-efi 3.0.15 is a hard dependency for NX then.

We're mandating NX enabled going forward as Microsoft won't sign shims anymore without NX, we want to make sure that we are compatible.